### PR TITLE
SUBS-493 MG Update Payment Method Enhancement

### PR DESCRIPTION
**Pull request by eddieferrer**
* Update for update MG. Now updates billing agreement instead of recreating it. Small UI changes as well.

### CODE REVIEW

**Comment by eddieferrer** [comment]
I'm going to refactor this component so it's slightly more generic when I implement Auto Deposit Update payment method. Since it is essentially the same code.

File: src/graphql/mutation/braintreeUpdateSubscriptionPaymentMethod.graphql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+mutation braintreeUpdateSubscriptionPaymentMethod($paymentMethodNonce: String!) {
+	my {
+		updateAutoDeposit (
+			autoDeposit: {
+			},
```
Comment (mcstover) [inline src/graphql/mutation/braintreeUpdateSubscriptionPaymentMethod.graphql:5]: what are we updating?

File: src/components/MonthlyGood/MonthlyGoodDropInPaymentWrapper.vue [hunk 2]
```diff
@@ -105,60 +107,107 @@ export default {
...
-					const errorCode = _get(kivaBraintreeResponse, 'errors[0].code');
-					const errorMessage = _get(kivaBraintreeResponse, 'errors[0].message');
-					const standardErrorCode = `(Braintree error: ${errorCode})`;
-					const standardError = `There was an error processing your payment.
+			if (this.action === 'Update') {
+				// Apollo call to the query mutation
+				this.apollo.mutate({
+					mutation: braintreeUpdateSubscriptionPaymentMethod,
+					variables: {
+						paymentMethodNonce: nonce,
```
Comment (mcstover) [inline src/components/MonthlyGood/MonthlyGoodDropInPaymentWrapper.vue:115]: what are the other properties we're updating?

File: src/graphql/mutation/braintreeUpdateSubscriptionPaymentMethod.graphql (hunk 1 above)
Comment (eddieferrer) [reply src/graphql/mutation/braintreeUpdateSubscriptionPaymentMethod.graphql:5]: I got an API error if the autoDeposit object is missing, this mutation updates the payment method, and leaves the other params intact.

File: src/components/MonthlyGood/MonthlyGoodDropInPaymentWrapper.vue (hunk 2 above)
Comment (eddieferrer) [reply src/components/MonthlyGood/MonthlyGoodDropInPaymentWrapper.vue:115]: The other properties are handled outside of the braintree drop in. If you recall, the modal has an "update settings" section and an "update payment method" section. Each section is independent of each other with it's own submit button.

**Review by mcstover — APPROVED** [review]

**Pull request merged**